File: 2/reader/Matrix2DReader.cpp
#include "Matrix2DReader.h"
#include "../matrix/Matrix2D.h"
#include <fstream>

Matrix *Matrix2DReader::read(std::ifstream &is) {
    int matrixSize;
    is >> matrixSize;
    Matrix2D *matrix = new Matrix2D(matrixSize);
    for (int i = 0; i < matrixSize; i++) {
        for (int j = 0; j < matrixSize; j++) {
            double value;
            is >> value;
            matrix->put(i, j, value);
        }
    }
    return matrix;
}

File: 3/reader/Matrix2DReader.py
from reader.MatrixReader import MatrixReader
import matrix


class Matrix2DReader(MatrixReader):

    def read(self, input_stream):
        matrix_size = int(input_stream.readline())
        new_matrix = matrix.Matrix2D(matrix_size)
        for i in range(matrix_size):
            values = list(map(float, input_stream.readline().split()))
            for j in range(matrix_size):
                new_matrix.put(i, j, values[j])
        return new_matrix